src/server/v2.0/handler/base.go

```diff
@@ -101,13 +101,19 @@ func (b *BaseAPI) RequireProjectAccess(ctx context.Context, projectIDOrName inte
 
 // BuildQuery builds the query model according to the query string
 func (b *BaseAPI) BuildQuery(ctx context.Context, query *string, pageNumber, pageSize *int64) (*q.Query, error) {
+
 	var (
 		qs string
 		pn int64
 		ps int64
 	)
 	if query != nil {
 		qs = *query
+		if escapedQuery, err := url.QueryUnescape(qs); err == nil {
```

Comment:
How about put the logic into the function `pkg/q.Build()`, it's the entrypoint for parsing query string

Reply:
Realy very strange that before use the query , we needed to unescape it. Maybe there is something wrong.

Reply:
Generally, the pattern "a=b" in the query is ok and do not need be escaped.
But we use "q=a=b" in the query, "a=b" is the value, thus UI will encode it. @heww

Reply:
@ywk move it to `pkg/q.Build()`